presto-main/src/main/java/com/facebook/presto/sql/planner/plan/StatisticAggregations.java

```diff
@@ -25,31 +25,70 @@
 import com.google.common.collect.ImmutableList;
 import com.google.common.collect.ImmutableMap;
 
+import javax.annotation.concurrent.Immutable;
+
 import java.util.List;
 import java.util.Map;
 import java.util.Optional;
+import java.util.stream.Collectors;
 
 import static com.google.common.base.Preconditions.checkArgument;
 import static java.util.Objects.requireNonNull;
 
 public class StatisticAggregations
 {
+    // Added to preserve the order for serializing and deserializing.
+    // map in c++ worker doesn't represent insertion order.
+    private final List<VariableReferenceExpression> variables;
+    private final List<Aggregation> variableAggregations;
```

Comment:
'variableAggregations' name doesn't match the value; why 'variable'?

I see that name 'aggregations' is used by the Map variable. I wonder whether we need Map variable at all? Quickly checking some usages, it looks like Map is not needed and a list would work just fine. If we do need it, maybe we can rename the Map variable to variableToAggregationMap or similar.

Alternatively, maybe we can drop the new List<Aggregation> variable and use just List<VariableReferenceExpression> + the original map. This should work and may be less hacky.